scipy/optimize/_minimize.py

```diff
@@ -531,29 +535,25 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
              RuntimeWarning)
 
     # check gradient vector
-    if meth == 'trust-constr':
-        if type(jac) is bool:
-            if jac:
-                fun = MemoizeJac(fun)
-                jac = fun.derivative
-            else:
-                jac = '2-point'
-        elif jac is None:
-            jac = '2-point'
-        elif not callable(jac) and jac not in ('2-point', '3-point', 'cs'):
-            raise ValueError("Unsupported jac definition.")
+    if callable(jac):
+        pass
+    elif jac is True:
+        # fun returns func and grad
+        fun = MemoizeJac(fun)
+        jac = fun.derivative
+    elif (jac in ('2-point', '3-point', 'cs') and
```

Comment:
Maybe add a constant tuple of allowed FD methods to _numdiff.py? This way, if a new scheme is added to approx_derivative at some point, we won't need to update it here

Reply:
This already [exists](https://github.com/scipy/scipy/blob/master/scipy/optimize/_differentiable_functions.py#L9), can make the change.

Reply:
Done.